File: mobile/lib/models/notification_settings.dart
class NotificationSettings {
  NotificationSettings({
    required this.orderUpdates,
    required this.promotions,
    required this.newProducts,
  });

  factory NotificationSettings.fromJson(Map<String, dynamic> json) {
    return NotificationSettings(
      orderUpdates: json['orderUpdates'],
      promotions: json['promotions'],
      newProducts: json['newProducts'],
    );
  }

  bool orderUpdates;
  bool promotions;
  bool newProducts;

  Map<String, dynamic> toJson() {
    return {
      'orderUpdates': orderUpdates,
      'promotions': promotions,
      'newProducts': newProducts,
    };
  }
}

File: src/Talabi.Core/Entities/NotificationSettings.cs
namespace Talabi.Core.Entities;

public class NotificationSettings : BaseEntity
{
    public string UserId { get; set; } = string.Empty;
    public AppUser? User { get; set; }
    
    public bool OrderUpdates { get; set; } = true;
    public bool Promotions { get; set; } = true;
    public bool NewProducts { get; set; } = true;
}
